### PR TITLE
Working solution for Avalonia UI Framework

### PR DESCRIPTION
**Pull request by mireo91**
**Idea**
Working solution to embed mpv into AvaloniaUI Framework - i needed linux platform to work. 
**Changes**
Mpv.NET changes:
- added cross-platform solution to dllLoadLibrary. I wanted to use shared libraries (.so) (I left previous loaddll solution but it is not needed right now)

Mpv.NET.Avalonia:
- view assembly implementation for AvaloniaUI (hope this could be published as nuget package)

Mpv.NET.AvaloniaExample:
- simple example for avalonia usage project

### CODE REVIEW

**Review by hudec117 — APPROVED** [review]

**Comment by hudec117** [comment]
Will try and get this out in V2

**Comment by mysteryx93** [comment]
How did you integrate it, did you do the OpenGL API integration?

I've been keeping an eye on this for a while now, for MediaPlayerUI
https://github.com/mysteryx93/MediaPlayerUI.NET/

**Comment by hudec117** [comment]
OpenGL integration is a headache I'm not going to go down haha

Looks like for Avalonia he created a VideoView component. I'm not too familiar with it but it compiles on Linux on my end and runs fine so I'm satisfied with it.

**Comment by mysteryx93** [comment]
VideoView from VLC component? Has it been tested properly on Linux and MacOS, and within UserControls? Does it just "work to display" with high CPU usage or does it benefit from hardware acceleration?

Here they are discussing a couple of issues
https://stackoverflow.com/questions/68292849/avaloniaui-cannot-embed-videoview-control-libvlcsharp-in-a-usercontrol

Someone wrote a demo OpenGL integration for MPV here
https://github.com/shodo/MPVCore/blob/master/SDLMpv/SDLMpv/MPV.cs

**Comment by hudec117** [comment]
Nah I meant this file https://github.com/hudec117/Mpv.NET-lib-/pull/42/files#diff-cba71949b0209c7635a4e8d33b0e146c1d2d3c7f741d133db1502d6cd01a1a82

I've seen that and I'm open to someone else submitting a PR with OpenGL support. As long as the "legacy" embedding was still supported.

**Comment by mireo91** [comment]
@mysteryx93 @hudec117 Sorry for not hearing any respose from me till now.
@hudec117 nice to here that code is satisfied by you :)  
@mysteryx93 I did it some time ago because I needed lightweight solution like vlc view component - but I discovered that vlc couldn't get me what I want. 
On linux you can use shared library from vlc and you can use mpv.so which comes with mpv-player (accualy mpv is really powerfull library and for me even better than vlc .net solution). If i remember it well I was doing this mpv view on the same way as vlc was done for avalonia example but I needed to develop and learn a lot about mpv library configurations and ofc on this mpv.net lib.
Basically you can set anything according to offical mpv manual https://mpv.io/manual/stable/ but I didnt need any of controls for my project.

**Comment by mysteryx93** [comment]
I haven't mentioned VLC. This discussion is about MPV.

I think that from what I've read, the VLC integration wasn't optimal either (can't confirm, haven't investigated).

What I'm saying is that the best way to integrate MPV is via Open GL (to benefit from GPU acceleration instead of CPU rendering), and that someone did a sample integration [here](https://github.com/shodo/MPVCore/blob/master/SDLMpv/SDLMpv/MPV.cs). Can you look into integrating this solution? The code is actually relatively simple.
